### PR TITLE
fix 286

### LINKED ISSUE #286: Matching `historical` and `sspXYZ` using `df` with incomplete columns make the historical period disappear
### Setup Information

- xscen version: 0.7.3-beta
- Python version: 3.11.5
- Operating System: Linux mint

### Description

I parse a local directory to create a `df`. The following fields:
```   
"activity"
'bias_adjust_institution'
'bias_adjust_project'
```
are unspecified, which causes the historical period to disappear when I proceed to make a catalog that matches `hist` and `fut`.

### Steps To Reproduce

```python
from xscen.catutils import parse_directory

df = parse_directory(
    directories=["/home/eridup1/tank/etiages/CMIP6_ornl_gov/"],
    patterns=[
        "{variable}_{?}_{source}_{experiment}_{member}_{?grid}_{DATES}.nc"
    ],
    homogenous_info={
        "mip_era": "CMIP6",
        "type": "simulation",
        "institution": "our",
        "processing_level": "extracted",
        "xrfreq": "MS",
        "frequency": "mon",
        "domain": "global",
        # I need to fill some or all of these fields, else historical datasets just disappear with match_hist_and_fut
        "activity": ".",
        'bias_adjust_institution':".",
        'bias_adjust_project':".",
    },
    read_from_file=["variable", "date_start", "date_end"],
)
subcat = xs.ProjectCatalog.from_df(df)
ds_dict = xs.search_data_catalogs(subcat,  variables_and_freqs={"rsds": "MS", "rsus":"MS"}, match_hist_and_fut=True)
```

The elements in `ds_dict` only contain the historical period when I fill the fields specified above. Otherwise, the historical period is ignored.

If I remove `match_hist_and_fut=True`, the historical periods remain, and they're separated. I checked and both `sspXYZ` and `historical` have `[None,None,None]` for the 3 specified fields, so it doesn't seem to be because of mismatching fields, but really the presence of `None` rather than some random string that makes the difference.

### Additional context

_No response_

### Contribution

- [ ] I would be willing/able to open a Pull Request to address this bug.

### PR DESCRIPTION
**Pull request by juliettelavoie**
### Pull Request Checklist:
- [x] This PR addresses an already opened issue (for bug fixes / features)
    - This PR fixes #286 
- [ ] (If applicable) Documentation has been added / updated (for bug fixes / features).
- [ ] (If applicable) Tests have been added.
- [x] This PR does not seem to break the templates.
- [x] HISTORY.rst has been updated (with summary of main changes).
  - [x] Link to issue (:issue:`number`) and pull request (:pull:`number`) has been added.

### What kind of change does this PR introduce?

*  add warning for the matching of hist and fut without activity

### CODE REVIEW

**Comment by Zeitsperre** [comment]
There's an issue with the current conda version. Pushing a fix soon.

**Review by aulemahal — APPROVED** [review]
Seems enough for the user to avoid the issue!

File: xscen/extract.py [hunk 1]
```diff
@@ -1237,6 +1237,12 @@ def _dispatch_historical_to_future(
         hist = sdf[sdf.experiment == "historical"]
         if hist.empty:
             continue
+        if pd.isna(sdf.activity).any():
+            warnings.warn(
+                f"np.NaN was found in the activity column of {group}. The rows with np.NaN activity will be skipped."
+                "If you want them to be included in the matching historical and future matching, please put a valid activity (https://xscen.readthedocs.io/en/latest/columns.html)."
```
Comment (juliettelavoie) [inline xscen/extract.py:1243]:
```suggestion
                "If you want them to be included in the historical and future matching, please put a valid activity (https://xscen.readthedocs.io/en/latest/columns.html)."
```

**Review by coxipi — APPROVED** [review]
That's clear, thanks

**Pull request merged**